### PR TITLE
Fix prelu ref when a.ndim < 2

### PR DESCRIPTION
**Pull request by soulitzer**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89809

Fixes https://github.com/pytorch/pytorch/issues/89560

Previously the test case for "input is 1-D or scalar + weight is not scalar" did not exist; adding it introduced some failures:
- forward AD (fixed in this PR)
- vmap (filed https://github.com/pytorch/pytorch/issues/89895)
- ref/meta (fixed this PR, though this also regresses nvFuser support)

### CODE REVIEW

**Review by ngimel — APPROVED** [review]

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -453,8 +453,9 @@ def sample_inputs_prelu(op_info, device, dtype, requires_grad, **kwargs):
             weight_tensor = torch.tensor(weight, device=device, dtype=dtype, requires_grad=requires_grad)
             yield SampleInput(make_arg(shape), args=(weight_tensor,))
 
-        channel_size = shape[1] if len(shape) >= 2 else 1
-        yield SampleInput(make_arg(shape), args=(make_arg((channel_size,)),))
+        if len(shape) >= 1:
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:456]: Comment here for what's happening

File: torch/_refs/nn/functional/__init__.py [hunk 2]
```diff
@@ -960,7 +960,12 @@ def prelu(a: TensorLikeType, weight: TensorLikeType) -> TensorLikeType:
         f"ndim = {weight.ndim}",
     )
     if a.ndim == 0:
-        weight = weight[0] if weight.ndim == 1 else weight
+        # NB: This diverges from the current prelu which actually allows this case
```
Comment (mruberry) [inline torch/_refs/nn/functional/__init__.py:963]: Would you elaborate on what's happening in this case and what the issue with nvFuser is?

Comment (ngimel) [reply torch/_refs/nn/functional/__init__.py:963]: We should skip nvfuser test, or nvfuser can do something in their integration, but we shouldn't change general refs to deviate from eager semantics.

Comment (soulitzer) [reply torch/_refs/nn/functional/__init__.py:963]: The issue with nvfuser was that I'm trying to call `select`, `view_as`, `view`, etc, to get weight to be a scalar, but those aren't supported.

Comment (ngimel) [reply torch/_refs/nn/functional/__init__.py]: But that happens only when a.ndim == 0 and weight.ndim is not zero? It's fine for nvfuser to not support this case for now (they should fix it).

**Pull request closed without merging**